server-spi-private/src/main/java/org/keycloak/authentication/FormContext.java

```diff
@@ -58,14 +58,16 @@ public interface FormContext {
     AuthenticationExecutionModel getExecution();
 
     /**
-     * Current user attached to this flow.  It can return null if no uesr has been identified yet
+     * Current user attached to this flow.  It can return null if no user has been identified yet
      *
      * @return
      */
     UserModel getUser();
 
     /**
      * Attach a specific user to this flow.
+     * 
+     * If there was another user attached to this flow calling this method overrides the previous setting. 
```

Comment:
@ahus1  I think with your rework this should be removed as well, but I'm not a 100% sure.

```suggestion
```